### DISCUSSION TITLE
Dynamic IPv6 Prefix Delegation / Alias Support

### DISCUSSION THREAD

**Discussion by SimonFischer04** [netbox-community/netbox#14723] [Ideas]
Hi,

I get a dynamic prefix delegated from my ISP.
F.e. 2a0d:3344:188c:6b::/56

This prefix is further divided into vlans/subnets:
- 2a0d:3344:188c:6b05::/64 (vlan 5)
- 2a0d:3344:188c:6b08::/64 (vlan 8)
- ...

Then DHCPv6 (or SLAAC) ensures static Addresses (at least the "host" part).

So when my ISP changes the Prefix I would have to change all of the prefix, sub-prefixes, ip-ranges, ip-addresses, ...

I'm not quite sure how to best implement this, so I'm starting this discussion here.
Some ideas:
- allow IPs to only specify the host part and link them to a prefix 
   -  same logic for prefixes => sub-prefixes
- (maybe combined) with a general implementation of the concept of "Aliases" in netbox?

What do you think about it? Any other ideas how to solve this.

**Comment by candlerb** [comment]
Making IP addresses and prefixes relative to their enclosing prefix would be interesting, but would be a complete redesign of the Netbox data model.

For now, I suggest you just use a script to renumber all your prefixes, addresses and ranges. I contributed one, but it hasn't been merged:
https://github.com/netbox-community/customizations/pull/94/files
(You just give it the old and new /56's, and it renumbers all addresses, prefixes and ranges contained within the old one to the new one)

**Reply by SimonFischer04** [reply]
@candlerb Yeah was already thinking it would not be to easy to implement. That was the reason for asking / discussing, maybe someone has an idea without changing that much in netbox.

Wow, thanks for this great script, good start in the meantime. Would be nice if there would also be an option to change the Aggregate.

**Reply by candlerb** [reply]
Netbox "Aggregates" are intended for blocks allocated from RIRs, I'd say that automated renumbering isn't normally required - but you could add that easily enough if you need it.

**Reply by SimonFischer04** [reply]
If my ISP is the one assigning me dynamic prefixes isn't then my ISP kind of an RIR? Or what would you define as the aggregate / how would you model this?
